### PR TITLE
[Improve](performance)Add a new schema_cache type schema_pb to cache …

### PR DESCRIPTION
**Pull request by qidaye**
…schema protobuf objs

## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by qidaye** [comment]
run buildall

File: be/src/olap/rowset/rowset_meta.cpp [hunk 1]
```diff
@@ -94,7 +97,29 @@ void RowsetMeta::set_fs(io::FileSystemSPtr fs) {
 void RowsetMeta::to_rowset_pb(RowsetMetaPB* rs_meta_pb) const {
     *rs_meta_pb = _rowset_meta_pb;
     if (_schema) {
-        _schema->to_schema_pb(rs_meta_pb->mutable_tablet_schema());
+        std::string schema_pb_key;
+        std::shared_ptr<TabletSchemaPB> cached_schema_pb;
+        // we cannot use schema_pb cache in variant schema or schema_version not set
+        if (_schema->num_variant_columns() == 0 && _schema->schema_version() > 0) {
```
Comment (eldenmoon) [inline be/src/olap/rowset/rowset_meta.cpp:103]: maybe we could capsulate the logical whether to use schema cache or not in a schema_cache function

File: be/src/olap/tablet_meta.cpp [hunk 2]
```diff
@@ -633,7 +638,30 @@ void TabletMeta::to_meta_pb(TabletMetaPB* tablet_meta_pb) {
     for (auto rs : _stale_rs_metas) {
         rs->to_rowset_pb(tablet_meta_pb->add_stale_rs_metas());
     }
-    _schema->to_schema_pb(tablet_meta_pb->mutable_schema());
+
+    std::string schema_pb_key;
+    std::shared_ptr<TabletSchemaPB> cached_schema_pb;
+    // we cannot use schema_pb cache in variant schema or schema_version not set
+    if (_schema->num_variant_columns() == 0 && _schema->schema_version() > 0) {
```
Comment (eldenmoon) [inline be/src/olap/tablet_meta.cpp:645]: diplicated logic?

**Review by yiguolei — CHANGES_REQUESTED** [review]
Do not add any cache now. There are already so many caches, please reuse them. It is too hard to control the memory.

**Pull request closed without merging**